lib/zipkin-tracer/zipkin_tracer_base.rb

```diff
@@ -0,0 +1,62 @@
+require 'faraday'
+require 'finagle-thrift'
+require 'finagle-thrift/tracer'
+
+module Trace
+  # This class is a base for tracers sending information to Zipkin.
+  # It knows about zipkin types of annotations and send traces when the server
+  # is done with its request
+  # Traces dealing with zipkin should inherit from this class and implement the
+  # flush! method which actually sends the information
+  class ZipkinTracerBase < Tracer
+    TRACER_CATEGORY = "zipkin".freeze
+
+    def initialize(options={})
+      @traces_buffer = options[:traces_buffer] || raise(ArgumentError, 'A proper buffer must be setup for the Zipkin tracer')
```

Comment:
Fail is a keyword. Untouchable.
"raise" is in fact Kernel.raise. It can be overwritten, if this gem is inside some projects which has let's say instrumentation and each raise does some kind of special notification or something, we would not be able to use that.